Device listing and additional device info

jteich opened the pull request:
This PR adds device enumeration as well as additional device information including card, device and subdevice names and numbers. 
Also contains some code cleanup. 
Let me know what you think.
Thanks!
Joseph

Commit:
Device listing and additional device info

Commit:
Nulling Alsa error handler; corrects illegal-instruction error

Commit:
Added ALSA error-handler for device enumeration; updated README

jteich committed:
Merge pull request #1 from jteich/illegal-instruction-fix

Illegal instruction fix

jteich commented:
Updated the PR by correcting an issue where enumerating devices after getting device details would result in a illegal-instruction error.

dmooney65 commented:
Hi Joseph,

Thanks for this - looks great. I've tested it in Intel and ARM and all good.

I'd be happy to merge this if you could add a couple of simple tests. One each in tests.js and ciTests.js as a minimum, more of you think necessary. Take a look at the existing ones for guidance - they're pretty simple and don't assume anything about particular hardware, just that responses exist where they should and errors are OK.

Run 'npm run test' for the default tests that assume an audio card exists.
'npm run test:ci' run only agains the 'null' device so are OK to run on Travis.

Thanks,

Dom

Commit:
Added unit tests

Commit:
Corrected unit tests

Commit:
Additonal unit test corrections

Commit:
Futther unit test corrections

Commit:
Additional unit-test correction

jteich commented:
Hi Dom, thanks for the feedback. I have updated the PR with a few tests per your suggestion that should cover the additional properties and method that I added.
